File: 6_december/giftinside.js
/** @param {string[]} gifts
 *  @returns {boolean} True if the gift is inside the box
 */
function inBox(box) {
    return box.slice(1, -1).some(row => row.slice(1, -1).includes('*'));
}

let gifts = inBox([
    "###",
    "# #*",
    "# #",
    "###",
  ]) // ➞ true

console.log(gifts)

File: 6_december/giftinside.py
def in_box(box):
  content = box[1:-1]
  for row in content:
    for char in row[1:-1]:
      if char == '*':
        return True
  return False

gifts = in_box([
    "###",
    "# #*",
    "# #",
    "###",
  ])
print(gifts)